src/charm.py

```diff
@@ -35,75 +34,101 @@ def _on_install(self, event):
                 return
 
             # install dependencies
+            check_call(["sudo", "apt-get", "update", "-qq"])
             check_call(
                 ["sudo", "apt-get", "install", "-y", "apt-transport-https", "lsb-release", "wget"]
             )
 
-            # add apt repo to sources
-            release = check_output(["lsb_release", "-c", "-s"]).decode("utf-8").rstrip("\n")
-            ps = Popen(
-                [
-                    "echo",
-                    "deb https://packagecloud.io/timescale/timescaledb/ubuntu/ {} main".format(
-                        release
-                    ),
-                ],
-                stdout=PIPE,
-            )
-            check_call(
-                ["sudo", "tee", "/etc/apt/sources.list.d/timescaledb.list"], stdin=ps.stdout
-            )
-            ps.wait()
-
-            # add apt key
-            ps = Popen(
-                [
-                    "wget",
-                    "--quiet",
-                    "-O",
-                    "-",
-                    "https://packagecloud.io/timescale/timescaledb/gpgkey",
-                ],
-                stdout=PIPE,
-            )
-            check_call(["sudo", "apt-key", "add", "-"], stdin=ps.stdout)
-            ps.wait()
-
-            # install TimescaleDB
-            check_call(["sudo", "apt-get", "update", "-qq"])
-
-            if os.path.exists("/var/lib/postgresql/12"):
-                pgver = 12
-            elif os.path.exists("/var/lib/postgresql/14"):
-                pgver = 14
+            debs = ["loader-deb", "tools-deb", "deb"]
+            deb_paths = []
+
+            for i in range(len(debs)):
```

Comment:
Just `for deb in debs`?
And then `path = self.model.resources.fetch(deb)` ?
like `paths = [self.model.resources.fetch(deb) for deb in debs]`